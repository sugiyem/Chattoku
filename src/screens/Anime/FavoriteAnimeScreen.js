import React, { useEffect, useState } from "react";
import { Alert, Dimensions, StyleSheet } from "react-native";
import { Avatar, Icon, ListItem } from "react-native-elements";
import {
  Button,
  ButtonText,
  CenteredBoldText,
  ScrollContainer
} from "../../styles/GeneralStyles";
import FetchFavoriteAnime from "../../services/Anime/FetchFavoriteAnime";
import styled from "styled-components/native";

const FavoriteAnimeScreen = ({ navigation }) => {
  const [anime, setAnime] = useState([]);
  const [expand, setExpand] = useState(null);

  const goToRecommendationPage = (item) => {
    navigation.navigate("FavAnimeRecommendation", { anime: item });
  };

  const onRightClick = (index) => {
    if (expand === index) {
      setExpand(null);
    } else {
      setExpand(index);
    }
  };

  const RenderAccordion = ({ item }) => (
    <AccordionContainer onPress={() => goToRecommendationPage(item)}>
      <Icon name="search" size={30} color="green" />
      <AccordionText>Get similar anime</AccordionText>
    </AccordionContainer>
  );

  const RenderTab = ({ item }) => (
    <>
      <Avatar
        size="medium"
        source={{ uri: item.image }}
        containerStyle={styles.imageContainer}
      />
      <Title>{item.title}</Title>
    </>
  );

  useEffect(() => {
    return FetchFavoriteAnime({
      onSuccesfulFetch: (data) => {
        setAnime(data);
      },
      onFailure: (error) => {
        Alert.alert("Error", error.message);
      }
    });
  }, []);

  return (
    <ScrollContainer>
      <CenteredBoldText size="20px">Favorite Anime List</CenteredBoldText>

      <Button onPress={() => navigation.goBack()}>
        <ButtonText>Go Back</ButtonText>
      </Button>

      <MarginView>
        {anime.map((item, idx) => (
          <ListItem.Accordion
            key={idx}
            bottomDivider
            content={<RenderTab item={item} />}
            isExpanded={expand === idx}
            onPress={() => onRightClick(idx)}
            style={styles.card}
          >
            {expand === idx && <RenderAccordion item={item} />}
          </ListItem.Accordion>
        ))}
      </MarginView>
    </ScrollContainer>
  );
};

export default FavoriteAnimeScreen;

const AccordionContainer = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
  margin: 15px;
  padding: 20px;
  border-radius: 100px;
  border-width: 1px;
  border-color: black;
  background-color: whitesmoke;
`;

const AccordionText = styled.Text`
  font-size: 16px;
  align-items: center;
  margin-left: 10px;
`;

const MarginView = styled.View`
  margin: 20px 5px;
`;

const Title = styled.Text`
  flex: 1;
  margin-left: 10px;
  font-size: 18px;
`;

const width = Dimensions.get("screen").width;

const styles = StyleSheet.create({
  imageContainer: {
    marginRight: 10,
    width: width / 5,
    height: undefined,
    aspectRatio: 3 / 4
  },
  card: {
    borderColor: "black",
    borderWidth: 0.5
  }
});
