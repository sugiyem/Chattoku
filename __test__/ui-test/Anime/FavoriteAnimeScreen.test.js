import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react-native";
import FavoriteAnimeScreen from "../../../src/screens/Anime/FavoriteAnimeScreen";

const mockNavigate = jest.fn();
const mockSetState = jest.fn();
const mockBackNavigate = jest.fn();
const mockNavigation = {
  goBack: mockBackNavigate,
  navigate: mockNavigate
};
const mockData = [
  {
    demographics: [],
    genres: ["Action", "Adventure", "Fantasy"],
    id: 23321,
    image: "https://cdn.myanimelist.net/images/anime/5/68097.jpg",
    themes: ["Adult Cast", "Video Game"],
    title: "Log Horizon 2nd Season",
    url: "https://myanimelist.net/anime/23321/Log_Horizon_2nd_Season"
  },
  {
    demographics: [],
    genres: ["Action", "Fantasy", "Supernatural"],
    id: 37675,
    image: "https://cdn.myanimelist.net/images/anime/1511/93473.jpg",
    themes: ["Isekai", "Video Game"],
    title: "Overlord III",
    url: "https://myanimelist.net/anime/37675/Overlord_III"
  }
];

jest.mock("@react-navigation/core", () => ({
  useNavigation: () => mockNavigation
}));

jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: (initial) => [
    typeof initial === "boolean" ? true : mockData,
    mockSetState
  ],
  useEffect: () => {}
}));

describe("Test FavoriteAnimeScreen UI", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<FavoriteAnimeScreen navigation={mockNavigation} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Back Button Works", () => {
    const { getByTestId } = render(
      <FavoriteAnimeScreen navigation={mockNavigation} />
    );

    fireEvent.press(getByTestId("backButton"));
    expect(mockBackNavigate).toHaveBeenCalled();
  });

  it("Get Similar Anime Button Works", () => {
    const { getByTestId } = render(
      <FavoriteAnimeScreen navigation={mockNavigation} />
    );

    fireEvent.press(getByTestId("23321"));
    expect(mockNavigate).toHaveBeenCalledWith("FavAnimeRecommendation", {
      anime: mockData[0]
    });

    fireEvent.press(getByTestId("37675"));
    expect(mockNavigate).toHaveBeenCalledWith("FavAnimeRecommendation", {
      anime: mockData[1]
    });
  });
});
