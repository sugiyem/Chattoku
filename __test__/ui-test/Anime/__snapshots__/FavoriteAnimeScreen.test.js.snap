// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FavoriteAnimeScreen UI renders correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4287090411,
        4283256319,
      ],
      "endPoint": Array [
        1,
        1,
      ],
      "locations": undefined,
      "startPoint": Array [
        0,
        0,
      ],
    }
  }
  style={
    Object {
      "flex": 1,
      "padding": 5,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "padding": 5,
      }
    }
  >
    <View>
      <Text
        size="20px"
        style={
          Array [
            Object {
              "fontFamily": "Gill Sans",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Favorite Anime List
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#7FFFD4",
            "borderRadius": 10,
            "borderWidth": 1,
            "marginBottom": 5,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 5,
            "opacity": 1,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          }
        }
        testID="backButton"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#483d8b",
                "fontSize": 14,
                "textAlign": "center",
              },
            ]
          }
        >
          Go Back
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 20,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 20,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 0.5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderBottomWidth": 0.5,
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
              }
            }
            testID="padView"
          >
            <View
              style={
                Object {
                  "aspectRatio": 0.75,
                  "backgroundColor": "transparent",
                  "height": undefined,
                  "marginRight": 10,
                  "width": 150,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "black": "#242424",
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "grey": "rgba(0, 0, 0, 0.54)",
                        "primary": "#2196f3",
                        "searchBg": "#dcdce1",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "default": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "web": Object {
                        "error": "#ff190c",
                        "grey": "#393e42",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                    "white": "#ffffff",
                  },
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 1,
                    "overflow": "hidden",
                    "position": "relative",
                  }
                }
              >
                <Image
                  onLoad={[Function]}
                  source={
                    Object {
                      "uri": "https://cdn.myanimelist.net/images/anime/5/68097.jpg",
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "flex": 1,
                      "height": undefined,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": undefined,
                    }
                  }
                  testID="RNE__Image"
                  theme={
                    Object {
                      "colors": Object {
                        "black": "#242424",
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "grey": "rgba(0, 0, 0, 0.54)",
                            "primary": "#2196f3",
                            "searchBg": "#dcdce1",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "default": Object {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "web": Object {
                            "error": "#ff190c",
                            "grey": "#393e42",
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#ca71eb",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                        "white": "#ffffff",
                      },
                    }
                  }
                  transition={true}
                  transitionDuration={360}
                />
                <View
                  accessibilityElementsHidden={true}
                  collapsable={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#bdbdbd",
                        "flex": 1,
                        "height": undefined,
                        "justifyContent": "center",
                        "width": undefined,
                      }
                    }
                    testID="RNE__Image__placeholder"
                  />
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "height": undefined,
                      "width": undefined,
                    }
                  }
                  testID="RNE__Image__children__container"
                />
              </View>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontSize": 18,
                    "marginLeft": 10,
                  },
                ]
              }
            >
              Log Horizon 2nd Season
            </Text>
            <View
              style={
                Object {
                  "paddingLeft": 16,
                }
              }
            />
            <View
              collapsable={false}
              style={
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Object {
              "maxHeight": "0%",
              "opacity": 0,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "whitesmoke",
                "borderColor": "black",
                "borderRadius": 100,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 15,
                "marginLeft": 15,
                "marginRight": 15,
                "marginTop": 15,
                "opacity": 1,
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              }
            }
            testID="23321"
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "fontSize": 16,
                    "marginLeft": 10,
                  },
                ]
              }
            >
              Get similar anime
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 0.5,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderBottomWidth": 0.5,
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
              }
            }
            testID="padView"
          >
            <View
              style={
                Object {
                  "aspectRatio": 0.75,
                  "backgroundColor": "transparent",
                  "height": undefined,
                  "marginRight": 10,
                  "width": 150,
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "black": "#242424",
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "grey": "rgba(0, 0, 0, 0.54)",
                        "primary": "#2196f3",
                        "searchBg": "#dcdce1",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "default": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "web": Object {
                        "error": "#ff190c",
                        "grey": "#393e42",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                    "white": "#ffffff",
                  },
                }
              }
            >
              <View
                accessibilityIgnoresInvertColors={true}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 1,
                    "overflow": "hidden",
                    "position": "relative",
                  }
                }
              >
                <Image
                  onLoad={[Function]}
                  source={
                    Object {
                      "uri": "https://cdn.myanimelist.net/images/anime/1511/93473.jpg",
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "flex": 1,
                      "height": undefined,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": undefined,
                    }
                  }
                  testID="RNE__Image"
                  theme={
                    Object {
                      "colors": Object {
                        "black": "#242424",
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "grey": "rgba(0, 0, 0, 0.54)",
                            "primary": "#2196f3",
                            "searchBg": "#dcdce1",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "default": Object {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "web": Object {
                            "error": "#ff190c",
                            "grey": "#393e42",
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#ca71eb",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                        "white": "#ffffff",
                      },
                    }
                  }
                  transition={true}
                  transitionDuration={360}
                />
                <View
                  accessibilityElementsHidden={true}
                  collapsable={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#bdbdbd",
                        "flex": 1,
                        "height": undefined,
                        "justifyContent": "center",
                        "width": undefined,
                      }
                    }
                    testID="RNE__Image__placeholder"
                  />
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                      "height": undefined,
                      "width": undefined,
                    }
                  }
                  testID="RNE__Image__children__container"
                />
              </View>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontSize": 18,
                    "marginLeft": 10,
                  },
                ]
              }
            >
              Overlord III
            </Text>
            <View
              style={
                Object {
                  "paddingLeft": 16,
                }
              }
            />
            <View
              collapsable={false}
              style={
                Object {
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Object {
              "maxHeight": "0%",
              "opacity": 0,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "whitesmoke",
                "borderColor": "black",
                "borderRadius": 100,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 15,
                "marginLeft": 15,
                "marginRight": 15,
                "marginTop": 15,
                "opacity": 1,
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              }
            }
            testID="37675"
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "fontSize": 16,
                    "marginLeft": 10,
                  },
                ]
              }
            >
              Get similar anime
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</ViewManagerAdapter_ExpoLinearGradient>
`;
